components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -1180,6 +1325,24 @@ subroutine radiation_tend(state,   ptend,    pbuf,          cam_out, cam_in,  &
       ! Zero-array for cloud properties if not diagnosed by microphysics
       real(r8), target, dimension(pcols,pver) :: zeros
       real(r8), dimension(pcols,pver) :: c_cldf  ! Combined cloud/snow fraction
+ 
+      !needed for SPA
+      integer :: year, month, day, tod, next_month
+      real(r8) :: fraction_of_month
+      real(r8), pointer :: aerosol_optical_property(:,:,:,:)
+      real(r8), pointer :: aer_tau_bnd_lw_mon_1(:,:,:)
+      real(r8), pointer :: aer_tau_bnd_lw_mon_2(:,:,:)
+      real(r8), pointer :: aer_tau_bnd_sw_mon_1(:,:,:)
+      real(r8), pointer :: aer_tau_bnd_sw_mon_2(:,:,:)
+      real(r8), pointer :: aer_ssa_bnd_sw_mon_1(:,:,:)
+      real(r8), pointer :: aer_ssa_bnd_sw_mon_2(:,:,:)
+      real(r8), pointer :: aer_asm_bnd_sw_mon_1(:,:,:)
+      real(r8), pointer :: aer_asm_bnd_sw_mon_2(:,:,:)
+
+      real(r8), dimension(12):: days_per_month
+      nullify(aerosol_optical_property)
+      !fill days_per_month
+      days_per_month = (/31,28,31,30,31,30,31,31,30,31,30,31/)
```

Comment:
Maybe add a comment here that SPA optics is not compatible with Leap Year.  It goes without saying really, but if in sometime in the distant future someone does decide to add leap year in it's nice to give them a fighting chance to find all the changes they need to make.  Being able to grep "leap year" might be something that they do.

Reply:
Good point. It may be worth adding the same comment in the microphysics SPA code as well.